validphys2/src/validphys/commondataparser.py

```diff
@@ -40,22 +108,825 @@
+            ensure that there are always three columns (repeat the last one) in the kinematics
+
+        drop_minmax: bool
+            Drop the min and max value, necessary for legacy comparisons
+
+        Returns
+        -------
+        pd.DataFrame
+            a dataframe containing the kinematics
+        """
+        kinematics_file = self.path_kinematics
+        kinyaml = _quick_yaml_load(kinematics_file)
+
+        kin_dict = {}
+        for bin_index, dbin in enumerate(kinyaml["bins"], start=1):
+            for d in dbin.values():
+                if d["mid"] is None:
+                    d["mid"] = 0.5 * (d["max"] + d["min"])
+
+                if drop_minmax:
+                    # TODO: for now we are dropping min/max information since it didn't exist in the past
+                    d["min"] = None
+                    d["max"] = None
+                else:
+                    # If we are not dropping it, ensure that it has something!
+                    d["min"] = d["min"] if d.get("min") is not None else d["mid"]
+                    d["max"] = d["max"] if d.get("max") is not None else d["mid"]
+
+            # The old commondata always had 3 kinematic variables and the code sometimes
+            # relies on this fact
+            # Add a fake one at the end repeating the last one
+            if fill_to_three and (ncol := len(dbin)) < 3:
+                for i in range(3 - ncol):
+                    dbin[f"extra_{i}"] = d
+
+            kin_dict[bin_index] = pd.DataFrame(dbin).stack()
+
+        if len(kin_dict) != self.ndata:
+            raise ValueError(
+                f"The number of bins in {kinematics_file} does not match ndata={self.ndata}"
+            )
+
+        return pd.concat(kin_dict, axis=1, names=[_INDEX_NAME]).swaplevel(0, 1).T
+
+    # Properties inherited from parent
+    @property
+    def nnpdf_metadata(self):
+        return self._parent.nnpdf_metadata
+
+    @property
+    def setname(self):
+        return self._parent.setname
+
+    @property
+    def experiment(self):
+        return self.setname.split("_")[0]
+
+    @property
+    def process(self):
+        return self.setname.split("_")[1]
+
+    @property
+    def cm_energy(self):
+        return self.setname.split("_")[2]
+
+    @property
+    def name(self):
+        return f"{self.setname}_{self.observable_name}"
+
+    @property
+    def is_ported_dataset(self):
+        """Return True if this is an automatically ported dataset that has not been updated"""
+        if self.ported_from is None:
+            return False
+
+        # If it is using a legacy variant and has a ported_from field, then it is a ported one
+        if self.applied_variant is not None and self.applied_variant.startswith("legacy"):
+            return True
+
+        # If not using a legacy variant, we consider it ported if the kin variables are still k1,k2,k3
+        return "k1" in self.kinematic_coverage and "k2" in self.kinematic_coverage
```

Comment:
Maybe the following is more correct?
```suggestion
        return set(["k1", "k2", "k3"]) == set(self.kinematic_coverage)
```